.cypress/integration/trace_analytics_test/trace_analytics_services.spec.js

```diff
@@ -171,7 +172,75 @@ describe('Testing Service map', () => {
     cy.get('[data-text = "Duration"]').click();
     cy.contains('100');
     cy.get('.euiFormLabel.euiFormControlLayout__prepend').contains('Focus on').should('exist');
-    cy.get('[placeholder="Service name"]').focus().type('database{enter}');
+  });
+
+  it('Render the vis-network div and canvas', () => {
+    // Check the view where ServiceMap component is rendered
+    cy.get('.euiText.euiText--medium .panel-title').contains('Service map');
+    cy.get('.vis-network').should('exist');
+    cy.get('.vis-network canvas').should('exist');
+
+    // Check the canvas is not empty
+    cy.get('.vis-network canvas')
+      .should('have.attr', 'style')
+      .and('include', 'position: relative')
+      .and('include', 'touch-action: none')
+      .and('include', 'user-select: none')
+      .and('include', 'width: 100%')
+      .and('include', 'height: 100%');
+
+    cy.get('.vis-network canvas').should('have.attr', 'width').and('not.eq', '0');
+    cy.get('.vis-network canvas').should('have.attr', 'height').and('not.eq', '0');
+  });
+
+  it.only('Click on a node to see the details', () => {
+    cy.get('.euiText.euiText--medium .panel-title').contains('Service map');
+    cy.get('.vis-network canvas').should('exist');
+
+    // Focus on "order" by selecting the first option
+    cy.get('.euiFormLabel.euiFormControlLayout__prepend').contains('Focus on').should('exist');
+    cy.get('[placeholder="Service name"]').click();
+    cy.get('.euiSelectableList__list li').eq(0).click();
+
+    // ensure rendering is complete before node click, replace wait
+    cy.get('[data-text="Errors"]').click();
+    cy.contains('60%');
+    cy.get('[data-text="Duration"]').click();
+    cy.contains('100');
```

Comment:
I have two question on this: 

1) We usually follow one test per it. Looks like we are mixing the focus test and the node click to view details test. This helps us to pin point the issue in the code quickly. Should we keep them independent?

2) Without focus, Why is the click on node with the `x, y` coordinates not working? Is there a possibility in our current code that the graph rendering can change shape on page refresh?

Reply:
1. I don't think this can be avoided for this particular test as we need the behavior to work locally and on this test unit. If the canvas fails to render the nodes, having a focus will also have the same error so this test still serves its purpose. If focus was broken then this would also fail but the error would still be identifiable.

2. Without focus it is failing because the environment cypress is using is applying the seed differently. (Below screenshot showing its layout vrs the local). The click fails because there is no node at the location in the layout that the cypress version non local is running. The service map will always be the same for the user regardless of refresh. My change works as cypress and local have the same layout.

Local:
<img width="995" alt="LocalTest" src="https://github.com/user-attachments/assets/a20f489c-021c-4c57-8d73-31a0ae6f1e12">

Cypress (Taken from recording of the failure):
<img width="1274" alt="CypressTest" src="https://github.com/user-attachments/assets/64d0e238-7669-4153-9715-643c27155691">

Local and Cypress with Focus
<img width="902" alt="Screenshot 2024-12-07 at 4 21 29 PM" src="https://github.com/user-attachments/assets/bcfb1485-bafc-431a-95fe-d6613ec682d4">

Reply:
> Without focus it is failing because the environment cypress is using is applying the seed differently. 

Are we assuming that the cypress issue won't be replicated on user's end?  We need to understand why is cypress seeing a different graph.